spinnaker_graph_front_end/__init__.py

```diff
@@ -185,30 +217,36 @@ def stop_run():
 
 
 def read_xml_file(file_path):
-    """ Reads a xml file and translates it into an application graph and \
-        machine graph (if required).
+    """ Read an XML file containing a graph description and translate it into\
+        an application graph and machine graph (if required).
+
+    .. warning::
+        This is not officially supported functionality yet.
 
     :param str file_path: the file path in absolute form
     """
     logger.warning("This functionality is not yet supported")
     _sim().read_xml_file(file_path)
 
 
-def add_vertex(cell_class, cell_params, label=None, constraints=None):
+def add_vertex(cell_class, cell_params, label=None, constraints=()):
```

Comment:
Again, not sure if having constraints be an empty tuple is correct

Reply:
Right at the bottom of the vertex stack, in `ConstrainedObject`, we do:
```python
if constraints is not None:
    for constraint in constraints:
        self.add_constraint(constraint)
```
(Or something very very similar to that.) But we can pass an empty tuple down and have the `for` do nothing instead of needing the `None` check, which produces nicer type documentation in many places as `None` was only ever used as a stand-in for the empty list (and lists don't go in default parameters as they're mutable).

Reply:
It's a pattern we use a lot.